api/src/main/java/org/ednovo/gooru/domain/service/FolderServiceImpl.java

```
@@ -0,0 +1,165 @@
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+/////////////////////////////////////////////////////////////
+package org.ednovo.gooru.domain.service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.ednovo.gooru.application.util.SerializerUtil;
+import org.ednovo.gooru.core.api.model.Collection;
+import org.ednovo.gooru.core.api.model.User;
+import org.ednovo.gooru.core.application.util.BaseUtil;
+import org.ednovo.gooru.core.constant.ConstantProperties;
+import org.ednovo.gooru.core.constant.ParameterProperties;
+import org.ednovo.gooru.domain.service.redis.RedisService;
+import org.ednovo.gooru.domain.service.search.SearchResults;
+import org.ednovo.gooru.infrastructure.persistence.hibernate.CollectionRepository;
+import org.ednovo.gooru.infrastructure.persistence.hibernate.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class FolderServiceImpl extends BaseServiceImpl implements FolderService, ParameterProperties, ConstantProperties {
+
+	@Autowired
+	private CollectionRepository collectionRepository;
+
+	@Autowired
+	private UserRepository userRepository;
+
+	@Autowired
+	private RedisService redisService;
+
+	public final static String TOC = "toc";
+
+	@Override
+	public SearchResults<Map<String, Object>> getMyCollectionsToc(String gooruUid, Integer limit, Integer offset, String sharing, String collectionType, String orderBy) {
+		if (!BaseUtil.isUuid(gooruUid)) {
+			User user = this.getUserRepository().getUserByUserName(gooruUid, true);
+			gooruUid = user != null ? user.getPartyUid() : null;
```

Hi @sheebans ,

If gooruUid is null we can skip the process and throw the exception saying user not found.